### PR TITLE
MintValue

### PR DESCRIPTION
**Pull request by Unisay**
Closes https://github.com/IntersectMBO/plutus/issues/6445

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Changelog fragments have been written (if appropriate)
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [x] Targeting master unless this is a cherry-pick backport
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: plutus-ledger-api/test/Spec/V3/MintValue.hs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    scaleTestsBy 5 \(mintValue :: MintValue) ->
+      let Value minted = mintValueMinted mintValue
+       in all (all (> 0) . Map.elems) (Map.elems minted)
+
+test_MintValueBurnedIsPositive :: TestTree
+test_MintValueBurnedIsPositive =
+  testProperty "MintValue burned is positive" $
+    scaleTestsBy 5 \(mintValue :: MintValue) ->
+      let Value burned = mintValueBurned mintValue
+       in all (all (> 0) . Map.elems) (Map.elems burned)
```
Comment (ana-pantilie) [inline plutus-ledger-api/test/Spec/V3/MintValue.hs:63]: All of the tests in this module do not test the Plinth implementation of the `MintValue` functions, but the Haskell one. You need to `compile` them as Plinth programs, similar to what is done in `plutus-tx-plugin.../test/.../AssocMap`.

**Review by ana-pantilie — CHANGES_REQUESTED** [review]

File: plutus-ledger-api/test/Spec/V3/MintValue.hs (hunk 1 above)
Comment (Unisay) [reply plutus-ledger-api/test/Spec/V3/MintValue.hs:63]: @ana-pantilie do you have an example of property-based testing in Plinth ?

Comment (ana-pantilie) [reply plutus-ledger-api/test/Spec/V3/MintValue.hs:63]: Yes, in the `AssocMap/Spec` module in `plutus-tx-plugin`. If we're testing `Value` the same as you did here then I think that's wrong, but maybe @zliu41 can comment as well

Comment (Unisay) [reply plutus-ledger-api/test/Spec/V3/MintValue.hs:63]: We agreed to bring up the topic of property testing on-chain code during the technical discussion meeting. This is a concern that makes sense to approach uniformly across the codebase (QuickCheck/Haskell vs. Hedgehog/Plint vs. other options)

Comment (Unisay) [reply plutus-ledger-api/test/Spec/V3/MintValue.hs:63]: So I'll merge these test as is and rewrite them as well as the ones for Value using QuickCheck/CEK in a separate PR.

**Review by ana-pantilie — APPROVED** [review]
LGTM, thanks for the discussion about testing!

Will a subsequent PR also implement a data-backed version of `MintValue`? I think that will be necessary in order to provide a fully data-backed `ScriptContext` to users.

**Pull request merged**